### ISSUE TITLE
About the pretrained model.

### ISSUE THREAD

**Issue by JiamWu** [StomachCold/HCTransformers#3]
The training of the first layer of HCTransformer requires a lot of computational resource overhead. Limited by computing resources， I can only reproduce your results on a single Nvidia 3090 GPU , but the results are far worse than the reported. Could you please provide the pretrained checkpoint of the HCTransformer? Thanks.

**Comment by learn2phoenix** [comment]
Is there any plan to release the pretrained weights?

**Comment by ooolike** [comment]
Such a good job! But could you please release the pretrained models? Thank u very much~

**Comment by StomachCold** [comment]
Thanks for your attention! Now we have provided the pre-trained weights and sample command-line instructions for evaluation. If there are any mistakes, please feel free to inform us.

**Comment by Jamine-W** [comment]
> Thanks for your attention! Now we have provided the pre-trained weights and sample command-line instructions for evaluation. If there are any mistakes, please feel free to inform us.

Thanks very much!

**Comment by kylewangt** [comment]
> Thanks for your attention! Now we have provided the pre-trained weights and sample command-line instructions for evaluation. If there are any mistakes, please feel free to inform us.
thank you for your work! i have a question about pretiained_model as follow:
when  i use the pretrained_model to  eval  the miniimagenet data,i got the acc about 72%, the result 
![image](https://user-images.githubusercontent.com/44918749/177262996-5c2fc1b5-395c-4a31-baa5-44c448fb8cd9.png)
the results are far worse than the reported,where is the problem? thank you again.

**Comment by kylewangt** [comment]
> 

thank for your work

**Comment by StomachCold** [comment]
> thank you for your work! i have a question about pretiained_model as follow:
> when  i use the pretrained_model to  eval  the miniimagenet data,i got the acc about 72%, the result
> ![image](https://user-images.githubusercontent.com/44918749/177262996-5c2fc1b5-395c-4a31-baa5-44c448fb8cd9.png)
> the results are far worse than the reported,where is the problem? thank you again.

1. We have updated the checkpoints_pooling, you can download it now. 
2. Since we train three Transformers, features extracted from them are three times larger than those in the first stage. We have reported the best performance using **ONLY** the features extracted from the **second** stage in our paper, which can be manipulated in:
https://github.com/StomachCold/HCTransformers/blob/6e37cece1365a7fa6c158470195a349dd33c7d34/testCos.py#L44

Our default implementation concatenates all features and the corresponding result is: 73.51%
<img  width="555" src="https://user-images.githubusercontent.com/36076017/177278135-3dd1a027-53a4-41f9-bdda-384930c4a8d8.jpg">

When we specify to use features extracted  from the second stage, it matches the reported result in our paper: 74.74%
<img  width="555" src="https://user-images.githubusercontent.com/36076017/177277688-33cea23d-57bf-4dd0-b010-1e18b2927fa1.png">

**Comment by kylewangt** [comment]
> > thank you for your work! i have a question about pretiained_model as follow:
> > when  i use the pretrained_model to  eval  the miniimagenet data,i got the acc about 72%, the result
> > ![image](https://user-images.githubusercontent.com/44918749/177262996-5c2fc1b5-395c-4a31-baa5-44c448fb8cd9.png)
> > the results are far worse than the reported,where is the problem? thank you again.
> 
> 1. We have updated the checkpoints_pooling, you can download it now.
> 2. Since we train three Transformers, features extracted from them are three times larger than those in the first stage. We have reported the best performance using **ONLY** the features extracted from the **second** stage in our paper, which can be manipulated in:
>    https://github.com/StomachCold/HCTransformers/blob/6e37cece1365a7fa6c158470195a349dd33c7d34/testCos.py#L44
> 
> Our default implementation concatenates all features and the corresponding result is: 73.51% <img alt="" width="555" src="https://user-images.githubusercontent.com/36076017/177278135-3dd1a027-53a4-41f9-bdda-384930c4a8d8.jpg">
> 
> When we specify to use features extracted from the second stage, it matches the reported result in our paper: 74.74% <img alt="" width="555" src="https://user-images.githubusercontent.com/36076017/177277688-33cea23d-57bf-4dd0-b010-1e18b2927fa1.png">

thank you for your reply

**Comment by learn2phoenix** [comment]
> Thanks for your attention! Now we have provided the pre-trained weights and sample command-line instructions for evaluation. If there are any mistakes, please feel free to inform us.

@StomachCold Thanks a lot for sharing this !!

**Comment by kylewangt** [comment]
@StomachCold 请问如何复现使用 checkpoints_pooling在mini-imagenet上5-shot的结果？

**Comment by ghlili** [comment]
> @StomachCold 请问如何复现使用 checkpoints_pooling在mini-imagenet上5-shot的结果？

与S2M2一样 我们是仿照他们写的 一次抽5张图片就好了 具体细节麻烦可以看看代码

**Comment by ghlili** [comment]
> @StomachCold 请问如何复现使用 checkpoints_pooling在mini-imagenet上5-shot的结果？

--num_shots 设5

**Comment by kylewangt** [comment]
> 

你好，参数--num_shots 设置的是5，但是得到的结果如下：
![image](https://user-images.githubusercontent.com/44918749/177463325-1c2c0152-a7d0-4a73-8701-5c0e04d2514a.png)
不知道是什么原因？

**Comment by Jamine-W** [comment]
> > 
> 
> 你好，参数--num_shots 设置的是5， ![image](https://user-images.githubusercontent.com/44918749/177463592-94f36ed2-0eee-4fe7-8c6b-b7cd478e359d.png)
> 
> 但是得到的结果如下： ![image](https://user-images.githubusercontent.com/44918749/177463325-1c2c0152-a7d0-4a73-8701-5c0e04d2514a.png) 不知道是什么原因？

I have the same problem with you. Using the checkpoint "checkpoint_mini.pth", the 1-shot result is 60.61% ,the 5-shot result is 75.19%, which are worse than the reported. I'm sure I do not make any changes to the code and use the command lines as recommended. So strange. 
![image](https://user-images.githubusercontent.com/46603986/177567894-663d7ab2-f2d8-4ea3-9ebd-37fee9c6a78c.png)
![image](https://user-images.githubusercontent.com/46603986/177567516-4f9a420f-3f2e-4704-a9ba-dc43b5be9a80.png)

**Comment by kylewangt** [comment]
> 
do you have any idea?

**Comment by StomachCold** [comment]
@kylewangt @Jamine-W
We guess it is because the dataset you are using has an image resolution of 84, which is different from what we use (480), both for mini- and tiered-. Actually, our HCTransformers prefer to take images of the original resolution, and we resize them to 480 when generating mini- and tiered ImageNet. To reproduce our results, you may try to follow the guidelines in the [Datasets Section](https://github.com/StomachCold/HCTransformers#datasets) to generate a compatible version of 𝒎𝒊𝒏𝒊ImageNet and evaluate on it. Please feel free to inform us if you have any difficulties.

**Comment by kylewangt** [comment]
@StomachCold thank you

**Comment by skariel** [comment]
Hi, thanks for the code and the paper they are really useful.

I evaluated mini-imagenet 5-ways 1-shot with the folllowing command:
```
python eval_hct_pooling.py --arch vit_small --server mini_pooling --partition test --checkpoint_key student --ckp_path ~/data/models/mini_pooling  --pretrained_weights ~/data/models/checkpoints_first/checkpoint_mini.pth --num_shots 1 
```

and I am getting accuracy ~1% lower than what is in the paper (73.7% vs 74.7%). This is what I did:

1. cloned the repo
2. downloaded the pretrained weights
3. prepared mini-imagenet with size 480x480
4. changed `testCos.py` to use the 2nd stage features: `z_all = torch.from_numpy(np.array(z_all))[:,:,384:768] # second stage` (in line 42)

Did you use seed 99 for the paper? granted, 73.7% is a very good accuracy but I'm wondering if I'm missing something
